### ISSUE TITLE
Investigate adding a CI job to catch typos

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#3805]
This exists: https://github.com/crate-ci/typos

I personally would definitely like tooling to catch typos in docs and code comments ([like fixed here](https://github.com/rust-bitcoin/rust-bitcoin/pull/3803)). I'm hesitant to add too much additional boilerplate or maintainer overhead for typo checking identifiers though.

**Comment by apoelstra** [comment]
I'd be in favor of this, if somebody else is willing to set it up and teach it all the Bitcoin jargon.

**Comment by tcharding** [comment]
Yeah I'm not keen to teach it either. I was surprised to see a bunch of lines in the PR linked above that seemed familiar enough that I may have written them even though I have spell check configured in my editor for docs.

**Comment by dpc** [comment]
`typos` is written with a philosophy opposite to spell checkers. It has a list of misspellings, so false positives are very rare. In Fedimint we use all sorts of Bitcoin slang too, and our config file is rather empty: https://github.com/fedimint/fedimint/blob/master/.typos.toml

**Comment by tcharding** [comment]
wen Fedi funding a few hours a week for a junior dev to port all the nice things from there to `rust-bitcoin`?

**Comment by tcharding** [comment]
Oh wait, the Nix related PRs would be non-stop ...

**Comment by dpc** [comment]
Obviously it's even better with Nix, but you only need to install `typos` and call `typos` to run it. :D There's https://github.com/crate-ci/typos/blob/master/docs/github-action.md#github-action

**Comment by tcharding** [comment]
Lets see how it goes: https://github.com/rust-bitcoin/rust-bitcoin/pull/3809

I copied your config file and trimmed it down.
